### PR TITLE
Fix AVX512 constexpr issue with GCC 8.5

### PR DESCRIPTION
**Pull request by crtrott**
I am also changing the SNL workflow to make sure we test AVX512 guaranteed (the machine we are running on has AVX512)

### CODE REVIEW

File: simd/src/Kokkos_SIMD_AVX512.hpp [hunk 1]
```diff
@@ -30,6 +30,18 @@
...
+// FIXME: Some intrinsics we are using are not marked constexpr in GCC
+// prior to some version. 8.5 fails 9.5 works.
+#ifdef KOKKOS_COMPILER_GNU
+#if KOKKOS_COMPILER_GNU < 950
+#define KOKKOS_IMPL_SIMD_CONSTEXPR
+#endif
+#endif
+
+#ifndef KOKKOS_IMPL_SIMD_CONSTEXPR
+#define KOKKOS_IMPL_SIMD_CONSTEXPR constexpr
```
Comment (nliber) [inline simd/src/Kokkos_SIMD_AVX512.hpp:42]: Should we `#undef` this at the end of the file?

**Review by nliber — CHANGES_REQUESTED** [review]

**Review by nmm0 — APPROVED** [review]
Looks good to me once nevin's request is addressed

**Comment by ldh4** [comment]
The constexpr-change was made to all simd backends in #6683. Should changes in this PR be applied to Scalar, AVX2 and NEON as well?

File: simd/src/Kokkos_SIMD_AVX512.hpp [hunk 2]
```diff
@@ -30,6 +30,18 @@
...
+#endif
```
Comment (masterleinad) [inline simd/src/Kokkos_SIMD_AVX512.hpp:43]:
```suggestion
#if defined(KOKKOS_COMPILER_GNU) && KOKKOS_COMPILER_GNU < 950
#define KOKKOS_IMPL_SIMD_CONSTEXPR
#else
#define KOKKOS_IMPL_SIMD_CONSTEXPR constexpr
#endif
```

**Comment by masterleinad** [comment]
https://godbolt.org/z/vvnEWW5zb demonstrates that the issue is already resolved with 9.1. Removing the extra check for `none_of` shows that we would get a warning even with recent g++ versions, see https://godbolt.org/z/zTcG95cEz. Hence, I would drop the `constexpr` annotation altogether (unless we want to provide a `consteval` fall back).

File: simd/src/Kokkos_SIMD_AVX512.hpp (hunk 1 above)
Comment (dalg24) [reply simd/src/Kokkos_SIMD_AVX512.hpp:42]: Ping

Comment (masterleinad) [reply simd/src/Kokkos_SIMD_AVX512.hpp:42]: Depends on https://github.com/kokkos/kokkos/pull/7486.

Comment (ldh4) [reply simd/src/Kokkos_SIMD_AVX512.hpp:42]: `constexpr` changes won't be part of #7486 (I will cherry-pick 7445d06bc7a7239f66ca59dc135d989287150d40 for reliable AVX512 testing).
I plan on revisiting constexpr-ness of simd interfaces in a separate PR later.
So, I think this PR could be closed for now.

**Comment by masterleinad** [comment]
constexpr changes won't be part of https://github.com/kokkos/kokkos/pull/7486

**Pull request closed without merging**